vllm/model_executor/model_loader/gguf_loader.py

```diff
@@ -32,8 +33,18 @@ def __init__(self, load_config: LoadConfig):
     def _prepare_weights(self, model_name_or_path: str):
         if os.path.isfile(model_name_or_path):
             return model_name_or_path
+        # for raw HTTPS link
+        if model_name_or_path.startswith(("http://", "https://")):
+            return hf_hub_download(url=model_name_or_path)
```

Comment:
```suggestion
        if model_name_or_path.startswith(("http://", "https://")) and model_name_or_path.endswith(".gguf"):
            return hf_hub_download(url=model_name_or_path)
```
Have better to add a suffix check here.